### PR TITLE
[Draft/luci-interpreter]Update *.test.cpp in luci-interpreter

### PR DESCRIPTION
**Pull request by struss**
This commit Update *.test.cpp to test uint8 case on operations used in Resnet.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/Add.test.cpp [hunk 1]
```diff
@@ -27,57 +27,106 @@ namespace
...
+  kernel.configure();
+  kernel.execute();
 
-    EXPECT_THAT(extractTensorData<float>(output_tensor),
-                ::testing::ElementsAreArray(ArrayFloatNear(test_outputs[i], 0.0001f)))
-        << "With shape number " << i;
+  if (std::is_floating_point<T>::value)
+  {
+    EXPECT_THAT(extractTensorData<T>(output_tensor),
+                ElementsAreArray(ArrayFloatNear(output_data, kQuantizedTolerance)));
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/Add.test.cpp:87]: This is floating point test. Why do you use kQuantizedTolerance?

File: compiler/luci-interpreter/src/kernels/Add.test.cpp [hunk 2]
```diff
@@ -27,57 +27,106 @@ namespace
...
+    std::vector<T> quantized_input1_value =
+        quantize<T>(input1_data, input_quant_param.first, input_quant_param.second);
+    std::vector<T> quantized_input2_value =
+        quantize<T>(input2_data, input_quant_param.first, input_quant_param.second);
+    input1_tensor.writeData(quantized_input1_value.data(),
+                            quantized_input1_value.size() * sizeof(T));
+    input2_tensor.writeData(quantized_input2_value.data(),
+                            quantized_input2_value.size() * sizeof(T));
+  }
+  std::pair<float, int32_t> output_quant_param = quantizationParams<T>(0, max * 2);
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/Add.test.cpp:73]: How did you get (0, max *2) ?

File: compiler/luci-interpreter/src/kernels/Add.test.cpp [hunk 3]
```diff
@@ -27,57 +27,106 @@ namespace
...
+           std::initializer_list<int32_t> output_shape, std::initializer_list<float> input1_data,
+           std::initializer_list<float> input2_data, std::initializer_list<float> output_data)
+{
+  float min, max;
+  min =
+      std::min(input1_data) > std::min(input2_data) ? std::min(input2_data) : std::min(input1_data);
+  max =
+      std::max(input1_data) > std::max(input2_data) ? std::max(input1_data) : std::max(input2_data);
+  float kQuantizedTolerance = GetTolerance(min, max);
+  std::pair<float, int32_t> input_quant_param = quantizationParams<T>(min, max);
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/Add.test.cpp:48]: Why do input1 and input2 have the same quantization parameter? Why not using separate quantization parameters? ex: input1_quant_param and input2_quant_param

**Comment by jinevening** [comment]
This PR is quite large, hence not easy to review at once. Can you upload PRs operator-by-operator?

I think it would be good to make an issue e.g., "[luci-interpreter] Test quantized kernels for ResNet-v1", and link the PRs to the issue.

**Comment by struss** [comment]
Okay! I will split into pieces of PR. Thanks!

File: compiler/luci-interpreter/src/kernels/Add.test.cpp [hunk 4]
```diff
@@ -27,57 +27,106 @@ namespace
...
+  else
   {
-    Tensor input1_tensor = makeInputTensor<DataType::FLOAT32>(test_shapes[i], input2_data);
-    Tensor input2_tensor = makeInputTensor<DataType::FLOAT32>(base_shape, input1_data);
-    Tensor output_tensor = makeOutputTensor(DataType::FLOAT32);
+    EXPECT_THAT(dequantize<T>(extractTensorData<T>(output_tensor), output_tensor.scale(),
+                              output_tensor.zero_point()),
+                ElementsAreArray(ArrayFloatNear(output_data, kQuantizedTolerance)));
+  }
+}
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/Add.test.cpp:95]: The previous test ran the test with exchanged inputs. Can you add the tests with exchanged inputs?

File: compiler/luci-interpreter/src/kernels/Add.test.cpp (hunk 1 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/Add.test.cpp:87]: Okay! I will update it!

File: compiler/luci-interpreter/src/kernels/Add.test.cpp (hunk 2 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/Add.test.cpp:73]: I just thought that summation of to tensor is below max*2 and 0 is minimal value of Relu function.

but, I thought sum of min/max value in each tensor would be better.

File: compiler/luci-interpreter/src/kernels/Add.test.cpp (hunk 3 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/Add.test.cpp:48]: Okay, I will update it!

**Comment by struss** [comment]
Parent Issue Done!

**Pull request closed without merging**